src/Util/FilterUtil.ts

```diff
@@ -71,6 +80,182 @@ class FilterUtil {
   //   return filter;
   // }
 
+  /**
+   * Checks if a feature matches the specified filter.
+   * Returns true if it matches, otherwise returns false.
+   */
+  static featureMatchesFilter = (filter: Filter, feature: any): boolean => {
+    if (filter.length === 0) {
+      return true;
+    }
+    let matchesFilter: boolean = true;
+    const operator: Operator = filter[0];
+    let isNestedFilter: boolean = false;
+    if (FilterUtil.operatorMapping[operator]) {
+      isNestedFilter = true;
+    }
+    if (isNestedFilter) {
+      switch (filter[0]) {
+        case '&&':
+          let intermediate = true;
+          let restFilter = filter.slice(1);
+          restFilter.forEach((f: Filter) => {
+            if (!FilterUtil.featureMatchesFilter(f, feature)) {
+              intermediate = false;
+            }
+          });
+          matchesFilter = intermediate;
+          break;
+        case '||':
+          intermediate = false;
+          restFilter = filter.slice(1);
+          restFilter.forEach((f: Filter) => {
+            if (FilterUtil.featureMatchesFilter(f, feature)) {
+              intermediate = true;
+            }
+          });
+          matchesFilter = intermediate;
+          break;
+        case '!':
+          matchesFilter = !FilterUtil.featureMatchesFilter(filter[1], feature);
+          break;
+        default:
+          throw new Error(`Cannot parse Filter. Unknown combination or negation operator.`);
+      }
+    } else {
+      const prop: any = _get(feature, 'properties[' + filter[1] + ']');
+      switch (filter[0]) {
+        case '==':
+          // tslint:disable-next-line
+          matchesFilter = prop == filter[2];
+          break;
+        case '*=':
+          if (prop && filter[2].length > prop.length) {
+            matchesFilter = false;
+          } else if (prop) {
+            matchesFilter = prop.indexOf(filter[2]) !== -1;
+          } else {
+            matchesFilter = false;
+          }
+          break;
+        case '!=':
+          // tslint:disable-next-line
+          matchesFilter = prop != filter[2];
+          break;
+        case '<':
+          if (typeof prop === typeof filter[2]) {
+            matchesFilter = prop < filter[2];
+          }
+          break;
+        case '<=':
+          if (typeof prop === typeof filter[2]) {
+            matchesFilter = prop <= filter[2];
+          }
+          break;
+        case '>':
+          if (typeof prop === typeof filter[2]) {
+            matchesFilter = prop > filter[2];
+          }
+          break;
+        case '>=':
+          if (typeof prop === typeof filter[2]) {
+            matchesFilter = prop >= filter[2];
+          }
+          break;
+        default:
+          throw new Error(`Cannot parse Filter. Unknown comparison operator.`);
+      }
+    }
+    return matchesFilter;
+  }
```

Comment:
It would be nice if you could split this up, this function is nearly 100 lines

Reply:
Done in https://github.com/terrestris/geostyler/pull/639/commits/788f981208c617a5791e71586aeb4629198cdbf5